docs: clarify reschedule, migrate, and replacement terminology (#24929)
hashicorp/nomad: website/content/docs/job-specification/migrate.mdx

within the job. Only service jobs with a count greater than 1 support migrate
blocks.

Migrating happens when a Nomad node is drained. When a node is lost, Nomad

> [replaces][] the allocations instead and the `migrate` block does not
> apply. When the agent fails to setup the allocation or the tasks of an

Reviewer: ```suggestion
[replaces][] the allocations instead and ignores the `migrate` block.
When the agent fails to set up the allocation or the tasks of an
```

allocation more than their [`restart`][] block allows, Nomad [reschedules][] the

> allocations instead and the `migrate` block does not apply.

Reviewer: ```suggestion
allocations instead and ignores the `migrate` block.
```
